torchmultimodal/modules/encoders/albef_text_encoder.py

```diff
@@ -0,0 +1,263 @@
+        for layer_module in self.layer:
+            hidden_states = layer_module(hidden_states, attention_mask)
+        return hidden_states
+
+
+class ALBEFTransformerLayer(nn.Module):
+    def __init__(
+        self,
+        hidden_size: int,
+        intermediate_size: int,
+        num_attention_heads: int,
+        layer_norm_eps: float,
+    ) -> None:
+        super().__init__()
+        self.attention = ALBEFTransformerAttention(
+            hidden_size, num_attention_heads, layer_norm_eps
+        )
+        self.dense1 = nn.Linear(hidden_size, intermediate_size)
+        self.transform_act_fn = nn.GELU()
+        self.dense2 = nn.Linear(intermediate_size, hidden_size)
+        self.LayerNorm = torch.nn.LayerNorm(hidden_size, eps=layer_norm_eps)
+
+    def forward(
+        self,
+        hidden_states: Tensor,
+        attention_mask: Optional[Tensor] = None,
+    ) -> Tensor:
+        attention_output = self.attention(hidden_states, attention_mask)
+        dense1_output = self.dense1(attention_output)
+        act_output = self.transform_act_fn(dense1_output)
+        dense2_output = self.dense2(act_output)
+        norm_output = self.LayerNorm(dense2_output + attention_output)
+        return norm_output
+
+
+class ALBEFTransformerAttention(nn.Module):
+    def __init__(
+        self,
+        hidden_size: int,
+        num_attention_heads: int,
+        layer_norm_eps: float,
+    ) -> None:
+        super().__init__()
+        self.self_attention = ALBEFTransformerSelfAttention(
+            hidden_size, num_attention_heads
+        )
+        self.dense = nn.Linear(hidden_size, hidden_size)
+        self.LayerNorm = torch.nn.LayerNorm(hidden_size, eps=layer_norm_eps)
+
+    def forward(
+        self,
+        hidden_states: Tensor,
+        attention_mask: Optional[Tensor] = None,
+    ) -> Tensor:
+        self_output = self.self_attention(hidden_states, attention_mask)
+        dense_output = self.dense(self_output)
+        attention_output = self.LayerNorm(dense_output + hidden_states)
+        return attention_output
+
+
+class ALBEFTransformerSelfAttention(nn.Module):
+    def __init__(
+        self,
+        hidden_size: int,
+        num_attention_heads: int,
+    ) -> None:
+        super().__init__()
+        if hidden_size % num_attention_heads != 0:
+            raise ValueError(
+                "The hidden size (%d) is not a multiple of the number of attention "
+                "heads (%d)" % (hidden_size, num_attention_heads)
+            )
+
+        self.num_attention_heads = num_attention_heads
+        self.attention_head_size = int(hidden_size / num_attention_heads)
+        self.all_head_size = self.num_attention_heads * self.attention_head_size
+
+        self.query = nn.Linear(hidden_size, self.all_head_size)
+        self.key = nn.Linear(hidden_size, self.all_head_size)
+        self.value = nn.Linear(hidden_size, self.all_head_size)
+
+    def transpose_for_scores(self, x: Tensor) -> Tensor:
```

Comment:
can we refactor this and the FLAVA `transpose_for_scores` into a utility so we don't have two copies of this?